### ISSUE TITLE
keywords are shared among all sql dialects

### ISSUE THREAD

**Issue by MrMeemus** [andialbrecht/sqlparse#384]
@andialbrecht , are there any plans in the near future to add support for other sql dialects without sharing keywords that might not be supported. For example right now the is_keyword function checks the dictionary for keywords in both PGSQL and Oracle, which would incorrectly mark a TSQL text as a token.keyword if it is found in PGSQL keywords.

What I imagine is in the parsestream call stack, to add a optional sql_dialect flag that would allow the parser to choose which SQL Keywords regex would be used. A unique Sql regex mapping to be added for which every additional sql dialect is supported.

Please let me know your thoughts. I would happily pick up this task with your support.

**Comment by andialbrecht** [comment]
@MrMeemus, yeah, it would be great to have a set of standard SQL keywords and additional dialect-specific keywords. This issue came up more often recently since the keyword list has expanded a bit.
The sql_dialect argument should be totally optional since the default behavior should be as dialect-agnostic as possible.

**Comment by MrMeemus** [comment]
@andialbrecht awesome!, glad to hear it's needed by users. Will have a PR out shortly :)

**Comment by ivot** [comment]
I started to look at this package to use it for Teradata and there are plenty of Teradata Specific Keywords I would like to add into scope, so having externally configurable list of Keywords is really benefitial.
